src/main/java/org/springframework/data/jpa/repository/query/JpaQueryExecution.java

```diff
@@ -204,13 +210,33 @@ protected Object doExecute(final AbstractJpaQuery repositoryQuery, final Object[
 		}
 
 		private long count(AbstractJpaQuery repositoryQuery, Object[] values) {
-
+			StringBuilder builder = new StringBuilder();
```

Comment:
To be honest I don't really grasp what is going on here.

I guess it starts with the previously existing version of the source code, which is bogus AFAIK since a count query must always return exactly one value.
Therefore this PR should be done on top of #400.

Any parsing of queries should happen where all the other parsing happens inside `AstractJpaQuery` derivatives.

And finally and possibly most crucial: The problem is that the generation of the count query is broken so that is the location that should get fixed.